azerothcore/azerothcore-wotlk
# fix(DB/conditions): Fix Keristrasza Gossip

## Changes Proposed:
-  Keristrasza now correctly changes gossip text based on quest progression
-  Adds missing gossip options to teleport player during quests
-  From TrinityCore: https://github.com/TrinityCore/TrinityCore/commit/30d1f470fbe9de4acfd3eaa22e52875c9531f0c5

## Issues Addressed:

- Closes issue like TrinityCore: https://github.com/TrinityCore/TrinityCore/issues/27663

## SOURCE:

https://www.youtube.com/watch?v=05EtuLo5kLE
https://github.com/TrinityCore/TrinityCore/commit/30d1f470fbe9de4acfd3eaa22e52875c9531f0c5

## Tests Performed:

- Before (dialog does not change - same as when NPC is released:)
![前](https://user-images.githubusercontent.com/98835050/152568021-bbb97fc3-3f27-4241-929f-104d33304fcd.PNG)

- After:
![后](https://user-images.githubusercontent.com/98835050/152568039-651e11b8-8056-448b-90eb-67851c00f0e7.PNG)

- New Gossip Options:
![1](https://user-images.githubusercontent.com/98835050/152568072-0809a5c4-1089-43f1-81c2-49bfa21f3d3e.PNG)

![2](https://user-images.githubusercontent.com/98835050/152568080-0430bebb-f4af-4cf1-9beb-6571d8636fa9.PNG)

## How to Test the Changes:


1. .additem 35671
2. Release NPC, check gossip
3. Change status for quest chain quests, see that gossip changes and options appear

## Known Issues and TODO List:


- The dialog options now appear correctly but do not function. They still need to be scripted.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
